std/digest/murmurhash.d

```diff
@@ -514,7 +514,15 @@ struct MurmurHash3(uint size /* 32 or 128 */ , uint opt = size_t.sizeof == 8 ? 6
         const remainderStart = numElements * Element.sizeof;
         foreach (ref const Element block; cast(const(Element[]))(data[0 .. remainderStart]))
         {
-            putElement(block);
+            version(ARM)
+            {
+                // Since the block may be unaligned, this can trip up ARM
+                // codegen, so place it in a newly aligned Element first.
+                Element alignedTemp = block;
```

Comment:
I'm actually surprised this apparently fixed the issue for LDC, it smells like wasted optimization opportunity somewhere. A rather compact proper (I hope, I just typed it blindly) fix with no optimizations for alignments of 4/2 bytes would be:

```D
version(ARM)
{
    // data.ptr appropriately aligned for type Element?
    if ((cast(size_t) data.ptr) & (Element.alignof - 1) == 0)
    {
        foreach (ref const Element block; cast(const(Element[]))(data[0 .. remainderStart]))
            putElement(block);
    }
    else
    {
        Element alignedBlock = void;
        for (size_t i = 0; i != remainderStart; i += Element.sizeof)
        {
            (cast(ubyte*) &alignedBlock)[0 .. Element.sizeof] = data[i .. i + Element.sizeof];
            putElement(alignedBlock);
        }
    }
}
else
{
    foreach (ref const Element block; cast(const(Element[]))(data[0 .. remainderStart]))
        putElement(block);
}
```